.github/workflows/release.yaml

```diff
@@ -0,0 +1,26 @@
+name: Release
```

Comment:
What's needed to make this work? I assume the version number in pyproject.toml needs to be updated too? Add a check to ensure pyproject.toml and the tag version match. Create a draft release with the name being the version.